src/Compilers/CSharp/Test/Syntax/Parsing/NullableParsingTests.cs

```diff
@@ -813,29 +806,15 @@ public void DeclarationPattern_NullableArray()
                 // (1,9): error CS0103: The name 'e' does not exist in the current context
                 // switch (e) { case T[]? t: break; }
                 Diagnostic(ErrorCode.ERR_NameNotInContext, "e").WithArguments("e").WithLocation(1, 9),
-                // (1,19): error CS8400: Feature 'type pattern' is not available in C# 8.0. Please use language version 9.0 or greater.
-                // switch (e) { case T[]? t: break; }
-                Diagnostic(ErrorCode.ERR_FeatureNotAvailableInVersion8, "T[]").WithArguments("type pattern", "9.0").WithLocation(1, 19),
                 // (1,19): error CS0246: The type or namespace name 'T' could not be found (are you missing a using directive or an assembly reference?)
                 // switch (e) { case T[]? t: break; }
                 Diagnostic(ErrorCode.ERR_SingleTypeNameNotFound, "T").WithArguments("T").WithLocation(1, 19),
-                // (1,22): error CS1003: Syntax error, ':' expected
-                // switch (e) { case T[]? t: break; }
-                Diagnostic(ErrorCode.ERR_SyntaxError, "?").WithArguments(":").WithLocation(1, 22),
-                // (1,22): error CS1513: } expected
-                // switch (e) { case T[]? t: break; }
-                Diagnostic(ErrorCode.ERR_RbraceExpected, "?").WithLocation(1, 22),
-                // (1,24): warning CS0164: This label has not been referenced
+                // (1,22): warning CS8632: The annotation for nullable reference types should only be used in code within a '#nullable' annotations context.
                 // switch (e) { case T[]? t: break; }
-                Diagnostic(ErrorCode.WRN_UnreferencedLabel, "t").WithLocation(1, 24));
+                Diagnostic(ErrorCode.WRN_MissingNonNullTypesContextForAnnotation, "?").WithLocation(1, 22));
```

Comment:
i would expect an error that says this is not legal as nullable types are not allowed in patterns.

Reply:
Added a bunch of binding tests with arrays and this case works a sexpected there. I assume, the error is not reported here because `T` is an error type

Reply:
Can you add a test where it is not an error type, to ensure we get an appropriate diagnostic

Reply:
There is already one: `PatternMatchingTests_NullableTypes`/`PatternsInSwitchStatement`. But it is with predefined `int?` type. Does it count?

Reply:
can you do it with a nullable reference type as well (specifically one with an identifier).

Reply:
Added tests of reference types in switch statements/expressions. Other cases (e.g. just `o is string?` and so on) are already covered in other tests